src/Io/Transaction.php

```diff
@@ -144,6 +142,30 @@ private function onResponseRedirect(ResponseInterface $response, RequestInterfac
         return $this->next($request);
     }
 
+    /**
+     * @param RequestInterface $request
+     * @param UriInterface $location
+     * @return \Clue\React\Buzz\Message\RequestInterface
+     */
+    private function makeRedirectRequest(RequestInterface $request, UriInterface $location)
+    {
+        $originalHost = $request->getUri()->getHost();
+        $request = $request
+            ->withoutHeader('Host')
+            ->withoutHeader('Content-Type')
+            ->withoutHeader('Content-Length');
+
+        // Remove authorization if changing hostnames (but not if just changing ports or protocols).
+        if ($location->getHost() !== $originalHost) {
```

Comment:
See my above comment, do you have a specific use case in mind where it makes sense to keep this if the URI components change, but the host stays the same? It's still my understanding that this should check the URI authority instead, but perhaps I'm missing a relevant use case?

Reply:
Actually my assumptions about this behavior are based on other implementations:
- https://github.com/request/request/pull/1184
- https://github.com/composer/composer/issues/6716

It looks like `request` and `composer` only check for host difference. Or I'm missing some-thing?

Reply:
Thank you for looking this up, no objections in this case 👍